src/recipe/application/mappers/recipe.mapper.ts

```diff
@@ -1,36 +1,40 @@
 import { RecipeEntity } from 'src/recipe/domain/recipe.entity';
 import {
-  RecipeIngredientDto,
+  RecipeItemDto,
   RecipeDto as RecipeUi,
 } from 'src/recipe/presentation/dto/recipe.dto';
 import { Unit } from 'src/constants/enums/unit.enum';
 import { Injectable } from '@nestjs/common';
 import {
-  RecipeIngredientEntity,
-  RecipeIngredientType,
-} from 'src/recipe/domain/recipe-ingredient.entity';
+  RecipeItemEntity,
+  RecipeItemType,
+} from 'src/recipe/domain/recipe-item.entity';
 import { RecipeDbRecord } from 'src/recipe/infra/repositories/prisma-recipe-repository.adapter';
 
 @Injectable()
 export class RecipeMapper {
   toDomain(record: RecipeDbRecord): RecipeEntity {
-    const recipeIngredients = record.ingredients.map((item) =>
-      RecipeIngredientEntity.create({
-        ingredientId: item.id,
-        name: item.ingredient?.name,
-        type: RecipeIngredientType.INGREDIENT,
-        quantity: item.quantity,
-        subRecipeId: null,
-      }),
+    const recipeIngredients = record.ingredients.map(
+      (item) =>
+        new RecipeItemEntity({
+          id: item.id,
+          props: {
+            name: item.ingredient?.name,
+            type: RecipeItemType.INGREDIENT,
+            quantity: item.quantity,
+          },
+        }),
     );
-    const recipeSubRecipes = record.usedAsSubRecipe.map((item) =>
-      RecipeIngredientEntity.create({
-        subRecipeId: item.id,
-        name: item.subRecipe?.name,
-        type: RecipeIngredientType.SUB_RECIPE,
-        quantity: item.quantity,
-        ingredientId: null,
-      }),
+    const recipeSubRecipes = record.usedAsSubRecipe.map(
```

Comment:
usedAsRecipe is array of recipeItems where subRecipeId is recipe itself and recipe id is recipe of parent recipe